account_operating_unit/models/account_move.py

```diff
@@ -170,16 +153,13 @@ def _onchange_operating_unit(self):
             for line in self.line_ids:
                 line.operating_unit_id = self.operating_unit_id
 
-    @api.onchange("journal_id")
-    def _onchange_journal(self):
-        if (
-            self.journal_id
-            and self.journal_id.operating_unit_id
-            and self.journal_id.operating_unit_id != self.operating_unit_id
-        ):
-            self.operating_unit_id = self.journal_id.operating_unit_id
-            for line in self.line_ids:
-                line.operating_unit_id = self.journal_id.operating_unit_id
+    @api.depends("journal_id")
+    def _compute_operating_unit(self):
+        for record in self:
+            if record.journal_id.operating_unit_id:
+                record.operating_unit_id = record.journal_id.operating_unit_id.id
```

Comment:
you don't need to explicitly assign an `id` to a many2one record, just `record.journal_id.operating_unit_id`

Reply:
Updated. Thanks for the review 👍